### DISCUSSION TITLE
changing val_check_interval from 0.01 to 0.005 changes number of steps in an epoch

### DISCUSSION THREAD

**Discussion by AyushP123** [Lightning-AI/pytorch-lightning#6440] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Changing val_check_interval from 0.01 to 0.005 changed the number of steps in an epoch from 2k to 3k for one of my experiments. Wanted to know if that is expected behavior.

**Comment by SkafteNicki** [comment, marked as answer]
The number of steps in a epoch that is shown in the progressbar is both the number of training steps AND the number of validation steps. Since `val_check_interval` changes the number of validation steps, it makes sense that you are seeing the number changing :]
